server/datastore/mysql/calendar_events.go

```diff
@@ -0,0 +1,150 @@
+package mysql
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/fleetdm/fleet/v4/server/contexts/ctxerr"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/jmoiron/sqlx"
+)
+
+func (ds *Datastore) NewCalendarEvent(
+	ctx context.Context,
+	email string,
+	startTime time.Time,
+	endTime time.Time,
+	data []byte,
+	hostID uint,
+) (*fleet.CalendarEvent, error) {
+	var calendarEvent *fleet.CalendarEvent
+	if err := ds.withRetryTxx(ctx, func(tx sqlx.ExtContext) error {
+		const calendarEventsQuery = `
+			INSERT INTO calendar_events (
+				email,
+				start_time,
+				end_time,
+				event
+			) VALUES (?, ?, ?, ?);
+		`
+		result, err := tx.ExecContext(
+			ctx,
+			calendarEventsQuery,
+			email,
+			startTime,
+			endTime,
+			data,
+		)
+		if err != nil {
+			return ctxerr.Wrap(ctx, err, "insert calendar event")
+		}
+
+		id, _ := result.LastInsertId()
+		calendarEvent = &fleet.CalendarEvent{
+			ID:        uint(id),
+			Email:     email,
+			StartTime: startTime,
+			EndTime:   endTime,
+			Data:      data,
+		}
+
+		const hostCalendarEventsQuery = `
+			INSERT INTO host_calendar_events (
+				host_id,
+				calendar_event_id,
+				webhook_status
+			) VALUES (?, ?, ?);
+		`
+		result, err = tx.ExecContext(
+			ctx,
+			hostCalendarEventsQuery,
+			hostID,
+			calendarEvent.ID,
+			fleet.CalendarWebhookStatusPending,
+		)
+		if err != nil {
+			return ctxerr.Wrap(ctx, err, "insert host calendar event")
+		}
+		return nil
+	}); err != nil {
+		return nil, ctxerr.Wrap(ctx, err)
+	}
+	return calendarEvent, nil
+}
+
+func (ds *Datastore) GetCalendarEvent(ctx context.Context, email string) (*fleet.CalendarEvent, error) {
+	const calendarEventsQuery = `
+		SELECT * FROM calendar_events WHERE email = ?;
```

Comment:
What if host is no longer linked to the email that was used to create the event?

Reply:
Good catch. Low probability bug (*), but we should probably add a `hostID`.

(*) Low probability because the email change would need to happen between the calls to `ds.GetHostsPolicyMemberships` and `ds.GetCalendarEvent`.

Reply:
`host_calendar_events` has the host id.

Reply:
> `host_calendar_events` has the host id.

Correct. Do you mean to check such host id in `host_calendar_events` would match a provided a `hostID uint` in this method?